src/components/pages/import-batch-view.js

```diff
@@ -0,0 +1,171 @@
+    <Table
+      key={task._id}
+      rowKey={attempt => `${task._id}_${attempt.startedOn}}`}
+      dataSource={task.attempts}
+      columns={attemtpsTableColumns}
+      pagination={false}
+      expandable={{
+        rowExpandable: attempt => attempt.errors?.length,
+        expandedRowRender: attempt => renderErrors(attempt)
+      }}
+      />
+  );
+
+  const renderHasErrrors = attempts => doesTaskHaveErrors(attempts) ? t('common:yes') : t('common:no');
+
+  const renderIsSuccessful = task => {
+    if (!task.processed) {
+      return <span />;
+    }
+
+    return isTaskSuccessful(task) ? t('common:yes') : t('common:no');
+  };
+
+  const taskTableColumns = [
+    { title: t('importType'), dataIndex: 'taskParams', width: '50px', render: renderImportType },
+    { title: t('documentTitle'), key: '_id', dataIndex: 'taskParams', width: '150px', render: renderTitle },
+    { title: t('taskStatus'), dataIndex: 'processed', width: '150px', render: renderStatus, sorter: task => task.processed ? 1 : -1 },
+    { title: t('hasErrors'), width: '150px', render: renderHasErrrors, sorter: task => doesTaskHaveErrors(task) ? 1 : -1 },
+    { title: t('isSuccessful'), width: '150px', render: renderIsSuccessful, sorter: task => isTaskSuccessful(task) ? 1 : -1 }
+  ];
+
+  return (
+    <Page>
+      <div className="ImportBatchViewPage">
+        <h1>{t('pageNames:importBatchView')}</h1>
+        <Row>
+          <Space>
+            <span>{t('batchStatus')}:</span>
+            <span>{renderBatchStatus(batch)}</span>
+          </Space>
+        </Row>
+        <Row>
+          <Space>
+            <span>{t('createdOn')}:</span>
+            <span>{renderDate(batch.createdOn)}</span>
+          </Space>
+        </Row>
+        <Row>
+          <Space>
+            <span>{t('completedOn')}:</span>
+            <span>{renderDate(batch.completedOn)}</span>
+          </Space>
+        </Row>
+        <Row>
+          <Space>
+            <span>{t('createdBy')}:</span>
+            <span>{batch.createdBy.username}</span>
+          </Space>
+        </Row>
+        <Collapse>
+          <Panel header={t('batchErrors')} extra={renderErrorCount(batch)}>
+            <List
+              dataSource={batch.errors}
+              renderItem={error => (
+                <List.Item>
+                  <span>{error}</span>
+                </List.Item>)}
+              />
+          </Panel>
+        </Collapse>
+        <Table
+          className="ImportBatchViewPage-tasksTable"
+          key={batch._id}
+          rowKey="_id"
+          dataSource={batch.tasks}
+          columns={taskTableColumns}
+          pagination={false}
+          expandable={{
+            expandedRowRender: attemptsRenderer,
+            rowExpandable: task => task.attempts?.length
+          }}
+          />
+      </div>
+    </Page>);
+}
+
+ImportBatchView.propTypes = {
+  initialState: PropTypes.shape({
+    batch: importBatchDetailsShape.isRequired
```

Comment:
There's a console warning about this line which I cannot figure out why it's there. The warning says that this is required but an undefined value was given. I'm not sure how that happens since we do server side rendering and the batch is always there when I send the page. The same happens on the import-batches.js

Reply:
![Untitled](https://user-images.githubusercontent.com/2676617/144396560-fc9aa1de-d1da-4128-884f-c834522972f9.png)

Reply:
Okay, I see, I don't know how to read, sorry.

Reply:
:laughing:

Reply:
Hmm, this seems to have been another one that I didn't see, I still get the original one. I'll try to learn to read now :)). I think I mislead you with this line, it was on the /import-batches/ page. I'm getting more and more confused with all the renaming and the code sorry.